src/run.ts

```diff
@@ -114,3 +115,42 @@ async function runSetup(): Promise<null | Error> {
   );
   return null;
 }
+
+function resolveVersionInput(): string {
+  const version = core.getInput("version");
+  const versionFilePath = core.getInput("go_version_file");
+
+  if (versionFilePath !== "") {
+    if (version !== "") {
+      core.warning(
+        "Both version and go_version_file inputs are specified, go_version_file will be preferred"
+      );
+    }
+
+    const goVersion = parseGoVersionFile(versionFilePath);
+    if (goVersion !== "") {
+      return goVersion;
+    }
+  }
+
+  return version;
+}
+
+function parseGoVersionFile(versionFilePath: string): string {
+  if (!fs.existsSync(versionFilePath)) {
+    throw new Error(
+      `The specified go version file: "${versionFilePath}" does not exist`
+    );
+  }
+
+  const contents = fs.readFileSync(versionFilePath).toString();
+  const basename = path.basename(versionFilePath);
+  if (basename === "go.mod" || basename === "go.work") {
+    const match = contents.match(
+      /^\s+github.com\/bufbuild\/buf v(\d+(?:\.\d+)*)/m
```

Comment:
Does this work with non-released versions?

Reply:
Likely not I fix and will add a test for that.